course7/homeworks/patrickkostjens/complex/CMakeLists.txt

```diff
@@ -2,4 +2,13 @@ cmake_minimum_required(VERSION 3.0)
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=c++14")
 
-add_executable(complex complex.cc)
+find_package(Boost REQUIRED COMPONENTS filesystem)
+
+INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
+
+add_library(print SHARED print.cc complex.cc complex.hh)
+SET_TARGET_PROPERTIES(print PROPERTIES PREFIX "")
+
+add_executable(complex main.cc calculator.hh calculator.cc complex.hh complex.cc)
+
+TARGET_LINK_LIBRARIES(complex LINK_PUBLIC ${Boost_LIBRARIES})
```

Comment:
On Linux, you'd have to link with "ld". You may want to practice detecting OS in CMake

Reply:
I'm not sure what you mean by this.